### PR TITLE
[WIP] SRAM usage optimization

### PR DESCRIPTION
**Pull request by leptun**
This really is [WIP]! Test it with caution. Anything could go wrong. A lot of changes have been made around the code and not really tested.
Each and every code change has to be checked that it still produces the same Serial output for backwards compatibility.

Resources used [MK3 FW3.9.0](https://travis-ci.com/github/prusa3d/Prusa-Firmware/builds/166731977) vs this [PR](https://travis-ci.com/github/prusa3d/Prusa-Firmware/builds/168728738):
|Printer| FW390 program space | FW390 mem | PR program space | PR mem|Diff program|Diff mem|
|-------|-------|-------|-------|-------|-------|-------| 
| MK3S | 248810 | 6486 | 248082 | 5835 | -728 | -651 |
| MK3 | 250396 | 6515 | 249538 | 5866 | -858 | -649 |
| MK25S 13a | 222264 | 6310 | 221612 | 5701 | -652 | -609 |
| MK25S 10a | 222144 | 6310 | 221118 | 5701 | -1026 | -609 |
| MK25 13a | 226274 | 6358 | 225442 | 5751 | -832 | -607 |
| MK25 10a | 226004 | 6358 | 225110 | 5751 | -894 | -607 |

PFW-1123

### CODE REVIEW

File: Firmware/ultralcd.cpp [hunk 1]
```diff
@@ -8725,26 +8666,17 @@ static int lcd_selftest_screen(TestScreen screen, int _progress, int _progress_s
 
 static void lcd_selftest_screen_step(int _row, int _col, int _state, const char *_name, const char *_indicator)
 {
-	lcd_set_cursor(_col, _row);
-
+	lcd_puts_at_P(_row, _col, _name);
```
Comment (ojrik) [inline Firmware/ultralcd.cpp]: lcd_puts_at_P(_col, _row, _name);

File: Firmware/ultralcd.cpp [hunk 2]
```diff
@@ -3397,7 +3388,7 @@ bool lcd_wait_for_pinda(float temp) {
 		lcd_set_cursor(0, 4);
 		lcd_print(LCD_STR_THERMOMETER[0]);
 		lcd_print(ftostr3(current_temperature_pinda));
-		lcd_print("/");
+		lcd_putc('>');
```
Comment (ojrik) [inline Firmware/ultralcd.cpp]: lcd_putc('/');

**Comment by leptun** [comment]
@ojrik Thank you for pointing out these typos. 👍

**Comment by DRracer** [comment]
@leptun before I forget - in case you experience performance issues
- LPM instruction takes 3 cycles (load from progmem)
- LDS takes 2 cycles (load from RAM)

**Pull request closed without merging**